ChangeLog.d/clean_pem_buffers.txt

```diff
@@ -0,0 +1,13 @@
+Bugfix
+  * As an optimisation to stop large buffers being written to the stack the
+    code was changed to write the raw der data to the same buffer that the
+    pem data would later be written to. Unfortunately even though the pem data
+    was always NULL terminated the der data would be left in the buffer after
+    the NULL termination. This is an issue as if you passed the same buffer
+    and length into one of the pem write functions and then into its
+    corresponding parse functions, then the parse function would fail, as its
+    test for whether the data was der or pem involved testing the last byte of
+    the buffer to see if it was zero. Tests have also been modified to try and
+    prevent this regression from happening again.
+    Thanks to Faless for reporting this in #3682
```

Comment:
We [no longer use changelog entries to acknowledge bug reporters](https://github.com/ARMmbed/mbedtls/pull/3726). And when we did, the wording was “Reported by NAME in #NUMBER.”